### PR TITLE
Add data source info to error message in TFRecord and Caffe parsers

### PR DESCRIPTION
**Pull request by jantonguirao**
Signed-off-by: Joaquin Anton <janton@nvidia.com>

#### Why we need this PR?
*Pick one, remove the rest*
- It adds additional info to an error message, which will be helpful to identify problems in a dataset

#### What happened in this PR?
*Fill relevant points, put NA otherwise. Replace anything inside []*
 - What solution was applied:
Print the source information of the sample and the total length of the raw data.
 - Affected modules and functionalities:
TFRecordParser/CaffeParser/Caffe2Parser 
- Key points relevant for the review:
All of it
 - Validation and testing:
N/A
 - Documentation (including examples):
N/A

**JIRA TASK**: *[Use DALI-XXXX or NA]*

### CODE REVIEW

**Comment by jantonguirao** [comment]
!build

File: dali/operators/reader/parser/tfrecord_parser.h [hunk 1]
```diff
@@ -58,13 +58,9 @@ class TFRecordParser : public Parser<Tensor<CPUBackend>> {
...
     // Omit length and crc
     raw_data = raw_data + sizeof(length) + sizeof(crc);
-    try {
-      DALI_ENFORCE(example.ParseFromArray(raw_data, length),
-          "Error in parsing - invalid TFRecord file!");
-    } catch (std::exception& e) {
-      std::string str = "Error while parsing TFRecord: " + std::string(e.what());
-      DALI_FAIL(str);
-    }
+    DALI_ENFORCE(example.ParseFromArray(raw_data, length),
```
Comment (JanuszL) [inline dali/operators/reader/parser/tfrecord_parser.h]: Could you double check if `ParseFromArray` cannot throw and we don't need this try-catch here?

Comment (jantonguirao) [reply dali/operators/reader/parser/tfrecord_parser.h]: The return code is used to signal an error. I was looking at the documentation and couldn't find anything about exceptions. Moreover, then I recalled that Google doesn't use C++ exceptions in their code: https://google.github.io/styleguide/cppguide.html#Exceptions

Comment (JanuszL) [reply dali/operators/reader/parser/tfrecord_parser.h]: Ok

File: dali/operators/reader/parser/caffe_parser.h [hunk 2]
```diff
@@ -30,7 +30,9 @@ class CaffeParser : public Parser<Tensor<CPUBackend>> {
   void Parse(const Tensor<CPUBackend>& data, SampleWorkspace* ws) override {
     caffe::Datum datum;
     int out_tensors = 0;
-    DALI_ENFORCE(datum.ParseFromArray(data.raw_data(), data.size()));
+    DALI_ENFORCE(datum.ParseFromArray(data.raw_data(), data.size()),
+      make_string("Error Caffe file: ", data.GetSourceInfo(),
```
Comment (klecki) [inline dali/operators/reader/parser/caffe_parser.h]:
```suggestion
      make_string("Error while parsing Caffe file: ", data.GetSourceInfo(),
```
?

File: dali/operators/reader/parser/tfrecord_parser.h [hunk 3]
```diff
@@ -58,13 +58,9 @@ class TFRecordParser : public Parser<Tensor<CPUBackend>> {
...
+      make_string("Error parsing TFRecord file: ", data.GetSourceInfo(),
```
Comment (klecki) [inline dali/operators/reader/parser/tfrecord_parser.h]:
```suggestion
      make_string("Error while parsing TFRecord file: ", data.GetSourceInfo(),
```

File: dali/operators/reader/parser/caffe2_parser.h [hunk 4]
```diff
@@ -189,7 +189,10 @@ class Caffe2Parser : public Parser<Tensor<CPUBackend>> {
     caffe2::TensorProtos protos;
     int consumed_inputs = 0;
 
-    DALI_ENFORCE(protos.ParseFromArray(data.data<uint8_t>(), data.size()));
+    DALI_ENFORCE(protos.ParseFromArray(data.data<uint8_t>(), data.size()),
+      make_string("Error parsing Caffe2 file: ", data.GetSourceInfo(),
```
Comment (klecki) [inline dali/operators/reader/parser/caffe2_parser.h]:
```suggestion
      make_string("Error while parsing Caffe2 file: ", data.GetSourceInfo(),
```

File: dali/operators/reader/parser/caffe2_parser.h [hunk 5]
```diff
@@ -189,7 +189,10 @@ class Caffe2Parser : public Parser<Tensor<CPUBackend>> {
...
+                  " (raw data length ", data.size(), ")"));
```
Comment (klecki) [inline dali/operators/reader/parser/caffe2_parser.h]: And maybe, but it's up to you:
```suggestion
                  " (raw data length: ", data.size(), " bytes)."));
```

**Comment by jantonguirao** [comment]
!build

**Pull request merged**